tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -1,5 +1,119 @@
+import cv2
 import numpy as np
+import scipy.ndimage as ndi
 from skimage import exposure, filters, morphology
```

Comment:
Not a hard requirements but cv2 morphology may be significantly faster than skimage morphology in some cases.

Reply:
Here morphology is only used to define neighbourhood for computing local entropy. skimage morphology is now replaced with the opencv function.

https://github.com/TissueImageAnalytics/tiatoolbox/blob/ee13a1f6f90b392278ac14b6c28d7d8465c083b7/tiatoolbox/tools/registration/wsi_registration.py#L165-L167